build-system/tasks/server-tests.js

```diff
@@ -49,12 +49,42 @@ async function getInput(inputFile) {
  * @return {string}
  */
 function getTestName(inputFile) {
-  const testPath = path.relative(transformsDir, inputFile);
-  const transformName = path.dirname(path.dirname(testPath));
-  const testSuffix = path.basename(testPath).replace('-input.html', '');
+  const transformName = path.basename(getTransformerDir(inputFile));
+  const testSuffix = getTestPath(inputFile);
   return `${transformName} → ${testSuffix}`;
 }
 
+/**
+ * Computes the directory of the transformer used in the test.
+ *
+ * @param {string} inputFile
+ * @return {string}
+ */
+function getTransformerDir(inputFile) {
+  // The prior assumption is that the transformer is in the parent directory.
+  // However, with Jest (and to mimic Jest), this is not necessarily true.
+  let transformerDir = inputFile;
+  while (path.basename(path.dirname(transformerDir)) != 'transforms') {
+    transformerDir = path.dirname(transformerDir);
```

Comment:
would you be able to break early so that you don't incur the next check?

Reply:
The way I'm thinking about approaching this would involve using another `if`, which can potentially double the number of total checks just for breaking early.
I'm not sure if that will be worthwhile.